pkg/provisioner.go

```diff
@@ -50,113 +51,94 @@ func (s *ProvisionerServer) ProvisionerCreateBucket(ctx context.Context,
 	req *cosi.ProvisionerDeleteBucketRequest) (*cosi.ProvisionerDeleteBucketResponse, error) {
-	/*	protocol := req.GetProtocol()
-		if protocol == nil {
-			klog.ErrorS(errors.New("Invalid Argument"), "Protocol is nil")
-			return nil, status.Error(codes.InvalidArgument, "Protocol is nil")
-		}
-		s3 := protocol.GetS3()
-		if s3 == nil {
-			klog.ErrorS(errors.New("Invalid Argument"), "S3 protocol is nil")
-			return nil, status.Error(codes.InvalidArgument, "S3 Protocol is nil")
-		}
+	klog.Infof("Delete bucket %q", req.GetBucketId())
+	if _, err := s.S3Client.DeleteBucket(req.GetBucketId()); err != nil {
+		klog.Info("failed to delete bucket %q", req.GetBucketId())
+		return nil, status.Error(codes.Internal, "Bucket deletion failed")
+	}
 
-		if _, err := s.S3Client.DeleteBucket(s3.BucketName); err != nil {
-			klog.Info("failed to delete bucket", s3.BucketName)
-			return nil, status.Error(codes.Internal, "Bucket deletion failed")
-		}
-	*/
 	return &cosi.ProvisionerDeleteBucketResponse{}, nil
 }
 
 func (s *ProvisionerServer) ProvisionerGrantBucketAccess(ctx context.Context,
 	req *cosi.ProvisionerGrantBucketAccessRequest) (*cosi.ProvisionerGrantBucketAccessResponse, error) {
-	/*	protocol := req.GetProtocol()
-		if protocol == nil {
-			klog.ErrorS(errors.New("Invalid Argument"), "Protocol is nil")
-			return nil, status.Error(codes.InvalidArgument, "Protocol is nil")
-		}
-		s3 := protocol.GetS3()
-		if s3 == nil {
-			klog.ErrorS(errors.New("Invalid Argument"), "S3 protocol is nil")
-			return nil, status.Error(codes.InvalidArgument, "S3 Protocol is nil")
-		}
-		user, err := s.radosgwAdminClient.CreateUser(radosgwapi.UserConfig{
-			UID:         req.GetAccountName(),
-			DisplayName: req.GetAccountName(),
-		})
-		if err != nil {
-			klog.Error("failed to create user", err)
-			return nil, status.Error(codes.Internal, "User creation failed")
-		}
-
-		policy, err := s.S3Client.GetBucketPolicy(s3.BucketName)
-		if err != nil {
-			if aerr, ok := err.(awserr.Error); ok {
-				if aerr.Code() != "NoSuchBucketPolicy" {
-					return nil, status.Error(codes.Internal, "fetching policy failed")
-				}
+	userName := req.GetAccountName()
+	bucketName := req.GetBucketId()
+	accessPolicy := req.GetAccessPolicy()
+	klog.Info("Granting user %q the policy %q to bucket %q", userName, bucketName, accessPolicy)
+	user, err := s.radosgwAdminClient.CreateUser(radosgwapi.UserConfig{
+		UID:         userName,
+		DisplayName: userName,
+	})
+	if err != nil {
+		klog.Error("failed to create user", err)
+		return nil, status.Error(codes.Internal, "User creation failed")
+	}
+	// TODO : Handle access policy in request, currently granting all perms to this user
+	policy, err := s.S3Client.GetBucketPolicy(bucketName)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			if aerr.Code() != "NoSuchBucketPolicy" {
+				return nil, status.Error(codes.Internal, "fetching policy failed")
 			}
 		}
+	}
```

Comment:
The end of the `if` block is reached when:
1. err == nil
2. err != nil and err cannot be cast to an aws error
3. err != nil, can be cast to an aws error and is == to NoSuchBucketPolicy.
Is this your intent?